### PR TITLE
Make searcher item names the same as node titles

### PR DESCRIPTION
**Pull request by brett-groshong**
### Purpose of this PR

This change makes it so that items in the searcher and node titles match, and are derived from `NodeUIDescriptor.DisplayName`.

---
### Testing status

This change can be tested by doing the following.
* Change a `NodeUIDescriptor.DisplayName`
* Wait for a domain reload
* Summon the searcher
* Notice: the updated display name is in the list
* Create a node from the item with the updated display name
* Notice: the node has the same title, as does the inspector for the node

---
### Comments to reviewers

* Took care of a TODO in `ShaderGraphSearcherDatabaseProvider` so that searcher items with conflicting names aren't added.
* Changed `ShaderGraphSearcherDatabaseProvider` to not use a deprecated constructor for `GraphNodeModelSearcherItem` and instead create them directly (removed interim class).

### CODE REVIEW

**Review by bencloward — APPROVED** [review]

**Review by Nightmask3 — CHANGES_REQUESTED** [review]
some of the node definitions look like they're incorrectly specifying their display names with dupes of other nodes, causing conflicts and for some of these nodes to be left out from the searcher

File: com.unity.sg2/Editor/GraphUI/DataModel/Searcher/ShaderGraphSearcherDatabaseProvider.cs [hunk 1]
```diff
@@ -35,53 +53,68 @@ static string GetCategoryPath(NodeUIDescriptor uiDescriptor)
...
+                        );
+                        string searcherItemName = uiHints.DisplayName;
+                        // fallback to the registry name if there is no display name
+                        if (string.IsNullOrEmpty(searcherItemName))
+                            searcherItemName = registryKey.Name;
+                        if (namesAddedToSearcher.Contains(searcherItemName))
+                        {
+                            // If there is already a SearcherItem with the current
+                            // name, warn and skip.
+                            Debug.LogWarning($"Not adding \"{searcherItemName}\" to the searcher. A searcher item with this name already exists.");
```
[the remark this answers is not in the document]
Comment (bencloward) [reply com.unity.sg2/Editor/GraphUI/DataModel/Searcher/ShaderGraphSearcherDatabaseProvider.cs:85]: Ouch - my bad.  Brett, you want to fix those as part of this set of changes, or should I change it separately?

**Review by Nightmask3 — APPROVED** [review]
Ben will address these issues in an immediate follow up PR

File: com.unity.sg2/Editor/GraphUI/DataModel/Searcher/ShaderGraphSearcherDatabaseProvider.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (brett-groshong) [reply com.unity.sg2/Editor/GraphUI/DataModel/Searcher/ShaderGraphSearcherDatabaseProvider.cs:86]: It looks like they were fixed in a separate PR. Closing this conversation and merging.

**Pull request merged**